src/renderer/components/database-item.jsx

```diff
@@ -0,0 +1,147 @@
+import React, { Component, PropTypes } from 'react';
+import TableSubmenu from './table-submenu.jsx';
+import { Menu, MenuItem } from 'remote';
+
+
+const STYLE = {
+  item: { wordBreak: 'break-all', cursor: 'default' },
+};
+
+
+export default class DatabaseItem extends Component {
+  static propTypes = {
+    database: PropTypes.object.isRequired,
+    item: PropTypes.object.isRequired,
+    dbObjectType: PropTypes.string.isRequired,
+    style: PropTypes.object,
+    columnsByTable: PropTypes.object,
+    triggersByTable: PropTypes.object,
+    onSelectItem: PropTypes.func,
+    onExecuteDefaultQuery: PropTypes.func,
+    onGetTableCreateScript: PropTypes.func,
+    onGetTableSelectScript: PropTypes.func,
+    onGetTableInsertScript: PropTypes.func,
+    onGetTableUpdateScript: PropTypes.func,
+    onGetTableDeleteScript: PropTypes.func,
+    onGetViewCreateScript: PropTypes.func,
+  }
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {};
+    this.contextMenu;
+  }
+
+  // Context menu is built dinamically on click (if it does not exist), because building
+  // menu onComponentDidMount or onComponentWillMount slows table listing when database
+  // has a loads of tables, because menu will be created (unnecessarily) for every table shown
+  onContextMenu(e){
+    e.preventDefault();
+    if (!this.contextMenu){
+      this.buildContextMenu();
+    }
+    this.contextMenu.popup(e.clientX, e.clientY);
+  }
+
+  buildContextMenu(){
+    const {
+      database,
+      item,
+      dbObjectType,
+      onExecuteDefaultQuery,
+      onGetTableCreateScript,
+      onGetTableSelectScript,
+      onGetTableInsertScript,
+      onGetTableUpdateScript,
+      onGetTableDeleteScript,
+      onGetViewCreateScript,
+    } = this.props;
+    const scriptFuncs = [onGetTableCreateScript, onGetTableSelectScript, onGetTableInsertScript, onGetTableUpdateScript, onGetTableDeleteScript];
+    const scriptsNames = ['CREATE', 'SELECT', 'INSERT', 'UPDATE', 'DELETE'];
+
+    this.contextMenu = new Menu();
+    if (dbObjectType === 'Tables' || dbObjectType === 'Views') {
```

Comment:
Could you avoid nested if and else statements. For instance this first if could be changed to leave the function. And the next if leave the function after appending the menu item.